test/test_zeroinput.c

```diff
@@ -0,0 +1,96 @@
+#include <assert.h>
```

Comment:
Please add some description about this test.

Reply:
Quick comment added. Let me know if you think it should be more in depth.

Reply:
I would prefer something like:
Test calling deflate with zero available input for each flush mode to check if the return value is equal to zlib's.

Reply:
But I'm ok if you think the actual comment is enough.